Add rw61 power management support

mmahadevan108 opened the pull request:
Switch the system timer to use OS Timer.
Added SoC Power Management support for RW61X. This PR supports Power Modes 1 and 2.

Tested with OS Timer wakeup using the test `zephyr/tests/subsys/pm/power_mgmt_soc`

Commit:
dts: arm: nxp_rw610: Add OS_Timer

Add OS Timer to device tree

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

mmahadevan108 committed:
dts: rw61x: Add Idle and Suspend power modes

The Power states map to Power Mode 1 and 2.

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

mmahadevan108 committed:
soc: rw6xx: Add power management support

Add support for Power modes 1 and 2.
The wakeup from power mode 2 is from the os timer.

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

mmahadevan108 committed:
boards: rd_rw61x: Switch to OS Timer for System clock

This clock is better suited for Power Management support
than Systick.

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

mmahadevan108 committed:
soc: nxp: Enable support for OS Timer on RW platform

The OS Timer will be used as the System Timer.

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

dleach02 commented on soc/nxp/rw/CMakeLists.txt:
```diff
@@ -7,6 +7,10 @@ zephyr_sources(
   )
 
+zephyr_sources_ifdef(CONFIG_PM
+  power.c
```
you are missing this file in the PR

dleach02 commented:
Which test case is used to test this and is there anything we need to do to ensure this platform is tested?

mmahadevan108 committed:
boards: rd_rw612: Update board documentation

Add information about OS Timer and Power Management support.

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

mmahadevan108 committed:
tests: pm: power_mgmt_soc: enable testing on rd_rw612_bga

Enable the SoC power management test suite on the NXP RW612 EVK
development board.

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

mmahadevan108 commented:
> zephyr/tests/subsys/pm/power_mgmt_soc

mmahadevan108 commented:
> Which test case is used to test this and is there anything we need to do to ensure this platform is tested?

`zephyr/tests/subsys/pm/power_mgmt_soc`. I have added rw_612 to the `testcase.yaml` file.
